std/php7/_std/Array.hx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C)2005-2016 Haxe Foundation
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+import php7.NativeIndexedArray;
+
+using php7.Global;
+
+
+// @:coreApi
+@:final
+class Array<T> implements php7.ArrayAccess<Int,T> {
+	public var length(default, null):Int;
+	var arr:NativeIndexedArray<T>;
+
+	public function new() {
+		arr = new NativeIndexedArray<T>();
+		length = 0;
+	}
+
+	public inline function concat(a:Array<T>):Array<T> {
+		return wrap(arr.merge(a.arr));
+	}
+
+	public inline function copy():Array<T> {
+		return wrap(arr);
+	}
+
+	public inline function filter(f:T->Bool):Array<T> {
+		return wrap(arr.filter(f));
+	}
+
+	public function indexOf(x:T, ?fromIndex:Int):Int {
```

Comment:
Would that be worthwhile?
I think most of the time it would be used with `fromIndex=null`, wouldn't it?
Theoretically `lastIndexOf` could be done with `array_search` as well but needs `krsort` two times. Not sure how that would perform.